streams_map.go

```diff
@@ -53,6 +58,8 @@ type streamsMap struct {
 	outgoingUniStreams  *outgoingUniStreamsMap
 	incomingBidiStreams *incomingBidiStreamsMap
 	incomingUniStreams  *incomingUniStreamsMap
+
+	resetErr utils.AtomicError // can be nil, otherwise must be an error
```

Comment:
I find it hard to understand which parts of the streamsMap are expected to be thread safe, probably some comments would help. 

That said, this looks weird to me: The error is set from `ResetFor0RTT` and read from the `Open*` / `Accept*` fns. So those are called from different threads? But then, they read the maps, which are set from `initMaps`, called from  `ResetFor0RTT`? Why does the error need to be protected, but not those maps?

Reply:
`Open` and `Accept` is called by the application. When 0-RTT is reset, these calls should return `Err0RTTRejected`, until the application calls `NextSession`.
`initMaps` is called after the error is set, so we don't need introduce a mutex for `Open` and `Accept`. Note that `ResetFor0RTT` also closes the maps, which makes them return an `Err0RTTRejected` if `Open` or `Accept` is called concurrently.